driver/userspace.c

```diff
@@ -145,42 +145,87 @@ WnbdSetInquiryData(_Inout_ PINQUIRYDATA InquiryData)
     WNBD_LOG_LOUD(": Exit");
 }
 
+#define MallocT(S) ExAllocatePoolWithTag(NonPagedPoolNx, S, 'pDBR')
+
 NTSTATUS
 WnbdInitializeScsiInfo(_In_ PSCSI_DEVICE_INFORMATION ScsiInfo)
 {
     WNBD_LOG_LOUD(": Enter");
     ASSERT(ScsiInfo);
-    HANDLE thread_handle;
+    HANDLE request_thread_handle = NULL, reply_thread_handle = NULL;
     NTSTATUS Status = STATUS_SUCCESS;
 
-    InitializeListHead(&ScsiInfo->ListHead);
-    KeInitializeSpinLock(&ScsiInfo->ListLock);
-    KeInitializeEvent(&ScsiInfo->DeviceEvent, SynchronizationEvent, FALSE);
+    InitializeListHead(&ScsiInfo->RequestListHead);
+    KeInitializeSpinLock(&ScsiInfo->RequestListLock);
+    // TODO: consider increasing the maximum value if this becomes configurable.
```

Comment:
Nit: Maybe we can remove the comment from here since there is one above where `MAX_IN_FLIGHT_REQUESTS` is defined.